index.js

```diff
@@ -72,35 +73,48 @@ const ACTIVITY_STATUS = "OK";
    * Must be a Github repository: Gitlab, Bitbucket, etc will not work properly
    */
-  extractGitHubRepoPath: (url, returnUrl = false) => {
-    if (!url) return "undefined_name";
+  extractGitHubRepoData: (url) => {
+    if (!url) return {
+      name: "undefined_name",
+      server: "undefined_name",
+      owner: "undefined_owner",
+    };
 
-    // Let's try to match https based repo url
-    let match = url.match(
-      /https?:\/\/(www\.)?(?<server>[\w.-]+).com\/(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/
-    );
-    if (match && match.groups && match.groups.owner && match.groups.name) 
-      return returnUrl ? `https://${match.groups.server}.com/${match.groups.owner}/${match.groups.name}` : `${match.groups.name}`;
+    const urlRegularExpressions = {
+      // https based repo url
+      'https': /https?:\/\/(www\.)?(?<server>[\w.-]+).com\/(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/,
 
-    // Now trying to match ssh based repo url
-    match = url.match(
-      /git@(?<server>[\w.-]+).com:(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/
-    );
-    if (match && match.groups && match.groups.owner && match.groups.name) 
-      return returnUrl ? `https://${match.groups.server}.com/${match.groups.owner}/${match.groups.name}` : `${match.groups.name}`;
+      // ssh based repo url
+      'ssh': /git@(?<server>[\w.-]+).com:(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/,
 
-    // Now trying to match http based git access
-    // Ref: https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#http-based-git-access-by-an-installation
-    match = url.match(
-      /https?:\/\/x\-access\-token\:(?<token>[\w.-]+)@github.com\/(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/
-    );
-    if (match && match.groups && match.groups.owner && match.groups.name) 
-      return returnUrl ? `https://github.com/${match.groups.owner}/${match.groups.name}` : `${match.groups.name}`;
-    else 
-      return null;
+      // x-access-token refeers to http based git access
+      // Ref: https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#http-based-git-access-by-an-installation
+      'x-access-token': /https?:\/\/x\-access\-token\:(?<token>[\w.-]+)@(?<server>[\w.-]+).com\/(?<owner>[\w.-]+)\/(?<name>[\w.-]+)\.git*/,
+    };
+
+    for (const regularExpression of Object.values(urlRegularExpressions)) {
+      const match = url.match(regularExpression);
+      if (!match || !match.groups) continue;
+      const { name, server, owner } = match.groups;
+      if (!name || !server || !owner) continue;
+
+      return { name, server, owner };
+    }
+
+    return null;
+  },
+
+  /**
+   * Format the given url (it can be with ssh or x-access-token format) to classic url like
+   * 'https://${server}.com/${owner}/${name}'
+   */
+  buildGitHubUrl: (url) => {
+    const { server, name, owner } = module.exports.extractGitHubRepoData(url);
+
+    return `https://${server}.com/${owner}/${name}`;
```

Comment:
Probably would be good to return null here as well... in the scenario in which `extractGitHubRepoData` could not parse the given URL, intead of returning a weird URL.

Reply:
Agreed